crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -0,0 +1,382 @@
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///
+///     Raises:
+///         FasterThanLightError: If speed is greater than the speed of light.
+///     """
+///     try:
+///         return distance / time
+///     except ZeroDivisionError as exc:
+///         raise FasterThanLightError from exc
+/// ```
+#[violation]
+pub struct DocstringMissingException {
+    id: String,
+}
+
+impl Violation for DocstringMissingException {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let DocstringMissingException { id } = self;
+        format!("Raised exception `{id}` missing from docstring")
+    }
+}
+
+/// ## What it does
+/// Checks for function docstrings that include exceptions which are not
+/// explicitly raised.
+///
+/// ## Why is this bad?
+/// Some conventions prefer non-explicit exceptions be omitted from the
+/// docstring.
+///
+/// ## Example
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///
+///     Raises:
+///         ZeroDivisionError: Divided by zero.
+///     """
+///     return distance / time
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def calculate_speed(distance: float, time: float) -> float:
+///     """Calculate speed as distance divided by time.
+///
+///     Args:
+///         distance: Distance traveled.
+///         time: Time spent traveling.
+///
+///     Returns:
+///         Speed as distance divided by time.
+///     """
+///     return distance / time
+/// ```
+#[violation]
+pub struct DocstringExtraneousException {
+    ids: Vec<String>,
+}
+
+impl Violation for DocstringExtraneousException {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let DocstringExtraneousException { ids } = self;
+
+        if let [id] = ids.as_slice() {
+            format!("Raised exception is not explicitly raised: `{id}`")
+        } else {
+            format!(
+                "Raised exceptions are not explicitly raised: {}",
+                ids.iter().map(|id| format!("`{id}`")).join(", ")
+            )
+        }
+    }
+}
+
+#[derive(Debug)]
+struct DocstringEntries<'a> {
+    raised_exceptions: Vec<QualifiedName<'a>>,
+    raised_exceptions_range: TextRange,
+}
+
+impl<'a> DocstringEntries<'a> {
+    /// Return the raised exceptions for the docstring, or `None` if the docstring does not contain
+    /// a `Raises` section.
+    fn from_sections(sections: &'a SectionContexts, style: SectionStyle) -> Option<Self> {
```

Comment:
I changed this to return `None` when there's no section, so we didn't have to `unwrap` the `raised_exceptions_range`. Feel free to revert in future PRs if that gets in the way, though I think something _like_ that is useful to get rid of the unwrap.